zoo/api/query.py

```diff
@@ -142,3 +153,80 @@ def resolve_all_dependency_usages(self, info, **kwargs):
         )
+
+    def resolve_all_search_results(self, info, **kwargs):
+        paginator = Paginator(**kwargs)
+        query_param = kwargs["search_query"] if "search_query" in kwargs else ""
+        search_type = (
+            kwargs["search_type"]
+            if "search_type" in kwargs
+            else IndexType.Service.value
+        )
+
+        indexes = meili_client.get_indexes()
+        # get all total counts for each meili index
+        search_index = None
+        total_count = {}
+        for index in indexes:
+            total_count[index["uid"]] = meili_client.get_index(index["uid"]).search(
+                query=query_param, opt_params={"offset": 0, "limit": 1}
+            )["nbHits"]
+
+            if index["uid"] == search_type:
+                search_index = index
+
+        # get all results for requested index
+        search_results = Query._get_all_for_index(
+            query_param, search_index, total_count[search_index["uid"]]
+        )
+        edges = []
+        for i, search_result in enumerate(
+            search_results[paginator.slice_from : paginator.slice_to]
+        ):
+            cursor = paginator.get_edge_cursor(i + 1)
+            edges.append(
+                types.SearchResultsConnection.Edge(node=search_result, cursor=cursor)
+            )
+
+        return types.SearchResultsConnection(
+            page_info=paginator.get_page_info(total_count[search_type]),
+            edges=edges,
+            total_services_count=total_count[IndexType.Service.value],
+            total_analytics_count=total_count[IndexType.Dependency.value],
+        )
+
+    @staticmethod
+    def _get_all_for_index(query_param, search_index, total_count):
+        offset = 0
+        limit = 100
+        search_results = []
+        while total_count != len(search_results):
+            results = meili_client.get_index(search_index["uid"]).search(
+                query=query_param, opt_params={"offset": offset, "limit": limit}
+            )
+
+            search_results = Query._objects_from_result(
+                results["hits"], search_index, search_results
+            )
+
+            offset += limit
+
+        return search_results
+
+    @staticmethod
+    def _objects_from_result(search_results, index, result_objects=None):
+        try:
+            model = apps.get_model(index["uid"], index["name"])
+            for i, result in enumerate(search_results):
+                object = model.objects.get(pk=result["id"])
+                if index["name"] == IndexType.Dependency.name:
+                    dependency = types.Dependency.from_db(object)
+                    node = types.SearchResult(id=i, dependency=dependency)
+                if index["name"] == IndexType.Service.name:
+                    service = types.Service.from_db(object)
+                    node = types.SearchResult(id=i, service=service)
+                result_objects.append(node)
```

Comment:
if `result_objects` is `None`, as it's specified on the default, this will fail 🤔 

overall I don't like the approach of calling this function with a list, it will cause unwanted side effects, like modifying the original array while also returning it, is this necessary?

just make it build and return a list without passing any pointer to it and handle merging lists wherever this is called